File: QF_Py/test/Submachine2/autocode/SubMachine2Impl.py
#
# Copyright 2009, 2010 California Institute of Technology.
# ALL RIGHTS RESERVED. U.S. Government Sponsorship acknowledged.
#
"""
File: SubMachine2Impl.py

Date Created:  05-Apr-2010 13:44:07
Created By:    scheng

Python custom-implementation class for functions referenced in
the SubMachine2 Statechart model.
"""
# Python imports here
import sys
import logging

# QF imports here
from qf import framework
from qf import event

# Submachine imports here
from SubMImpl import *


# Module globals initialized here
LOGGER = logging.getLogger('SubMachine2Logger')


def printf(format, *args):
    # sys.stdout.write(format % args)
    LOGGER.info(format % args)


class SubMachine2Impl(object):
    """
    SubMachine2 state machine implementation object.
    """
    def __init__(self, machine_name="SubMachine2"):
        """
        Constructor
        """
        self.__machine_name = machine_name
        self.fr = framework.QF.getInstance()
        self.activeObjects = self.fr.getActiveDict()
        #
        print "*Accessible: IMPL['SubMachine2']"

    def set(self, attr, value=True):
        """
        Sets the boolean value of an attribute, default to True
        """
        if type(value) == bool:
            self.__dict__['_'+attr] = value
            print attr, "set to", value
        else:
            print 'Value must be a bool, either "True" or "False" (no quotes)!'

    def get(self, attrname):
        """
        Gets the boolean value of an attribute
        """
        return self.__dict__['_'+attrname]

    def clear(self, attrname):
        """
        Clears the value of an attribute to False
        """
        self.__dict__['_'+attrname] = False



class S1_SubMImpl(SubMImpl):
    """
    S1_SubMImpl SubMachine implementation override object.

    Override the action methods of the SubMachine individually if custom
    behavior is desired.
    """
    def __init__(self, machine_name="S1:SubM"):
        """
        Override constructor
        """
        SubMImpl.__init__(self, machine_name)
        self.__machine_name = machine_name

    def takeAction(self, e):
        """
        Override implementation Action method for takeAction().
        """
        print "%s.takeAction() overridden action implementation invoked" % self.__machine_name


class S2_SubMImpl(SubMImpl):
    """
    S2_SubMImpl SubMachine implementation override object.

    Override the action methods of the SubMachine individually if custom
    behavior is desired.
    """
    def __init__(self, machine_name="S2:SubM"):
        """
        Override constructor
        """
        SubMImpl.__init__(self, machine_name)
        self.__machine_name = machine_name

    def takeAction(self, e):
        """
        Override implementation Action method for takeAction().
        """
        print "%s.takeAction() overridden action implementation invoked" % self.__machine_name

File: QF_Cpp/test/Submachine2/autocode/SubMachine2Impl.cpp
//-*- Mode: C++; -*-
//===========================================================================
// This software contains Caltech/JPL confidential information.
//
// Copyright 2009-2011, by the California Institute of Technology.
// ALL RIGHTS RESERVED. United States Government Sponsorship Acknowledged.
// Any commercial use must be negotiated with the Office of Technology
// Transfer at the California Institute of Technology.
//
// This software may be subject to US export control laws and
// regulations. By accepting this document, the user agrees to comply
// with all applicable U.S. export laws and regulations, including the
// International Traffic and Arms Regulations, 22 C.F.R. 120-130 and the
// Export Administration Regulations, 15 C.F.R. 730-744. User has the
// responsibility to obtain export licenses, or other export authority as
// may be required before exporting such information to foreign countries
// or providing access to foreign persons.
//===========================================================================
//
//       File: SubMachine2Impl.cpp
// Created on: 30-Jul-2011 10:19:33
//     Author: scheng@jpl.nasa.gov
//
// This file was stubbed by the JPL StateChart Autocoders, which converts UML
// Statecharts, in XML, to a C++ variant of Miro Samek's Quantum Framework.
//===========================================================================
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <log_event.h>
#include <qf_port.h>
#include <qassert.h>
#include <SubMachine2Impl.h>
#include <StatechartSignals.h>

/**
 * Default constructor.
 */
SubMachine2Impl::SubMachine2Impl () {
    strcpy(this->machineName, "SubMachine2");

    AttributeMapper::init(this);
}

/**
 * Required virtual destructor.
 */
SubMachine2Impl::~SubMachine2Impl () {
    AttributeMapper::clean(this);
}

void SubMachine2Impl::setQActive (QActive* active) {
    m_active = active;
}


////////////////////////////////////////////
// Action and guard implementation methods
////////////////////////////////////////////


/////////////////////////////////////////////////////////////////////////
// Override submachine-instance action and guard implementation methods
/////////////////////////////////////////////////////////////////////////

/**
 * Default constructor.
 */
S1_SubMImpl::S1_SubMImpl () {
    strcpy(this->machineName, "S1:SubM");

    AttributeMapper::init(this);
}

/**
 * Required virtual destructor.
 */
S1_SubMImpl::~S1_SubMImpl () {
    AttributeMapper::clean(this);
}

void S1_SubMImpl::takeAction () {
    printf("%s.takeAction() overridden action implementation invoked\n", this->machineName);
}

/**
 * Default constructor.
 */
S2_SubMImpl::S2_SubMImpl () {
    strcpy(this->machineName, "S2:SubM");

    AttributeMapper::init(this);
}

/**
 * Required virtual destructor.
 */
S2_SubMImpl::~S2_SubMImpl () {
    AttributeMapper::clean(this);
}

void S2_SubMImpl::takeAction () {
    printf("%s.takeAction() overridden action implementation invoked\n", this->machineName);
}
